code/game/objects/items/devices/scanners.dm

```diff
@@ -300,6 +301,12 @@ MASS SPECTROMETER
 		else
 			user << "<span class='info'>Plasma: [round(plasma_concentration*100, 0.01)] %</span>"
 
+		if(bz_concentration > 0.005)
+			user << "<span class='alert'>Hallucinogen: [round(plasma_concentration*100, 0.01)] %</span>"
+		else
+			user << "<span class='info'>Hallucinogen: [round(plasma_concentration*100, 0.01)] %</span>"
+

```

Comment:
Add var to gas datums that determines their "dangerous" level

In this proc, loop over gases, in the body of the loop check if the current level is as bad as the "dangerous" level set in their respective vars, based on that send user info in appropriate span class